include/scea/algo/bc.hpp

```diff
@@ -23,29 +23,23 @@ class BetweennessCentrality : public Algo {
       std::numeric_limits<uint64_t>::max() / 4;
   constexpr static const unsigned LEVEL_CHUNK_SIZE = 256u;
 
-  uint64_t numOfSources;
-  std::string sourceFile;
+  std::vector<uint64_t> sources; // empty if all vertices are sources
 
 public:
-  BetweennessCentrality(uint64_t _numOfSources  = 0,
-                        std::string _sourceFile = "")
-      : numOfSources(_numOfSources), sourceFile(_sourceFile) {}
-
-  static galois::LargeArray<double> compute(scea::graph::MutableGraph& g,
-                                            uint64_t numOfSources  = 0,
-                                            std::string sourceFile = "") {
-    std::vector<size_t> sources;
-    if (numOfSources == 0) {
-      if (sourceFile == "") {
-        sources.resize(g.size());
-        std::iota(sources.begin(), sources.end(), 0);
-      } else {
-        sources = sampleFromFile(sourceFile);
-      }
-    } else {
-      sources = sampleGen(0, g.size(), numOfSources, sourceFile);
+  BetweennessCentrality(uint64_t num_vertices, unsigned int rseed = 0,
+                        uint64_t num_src = 0) {
+    sources.resize(num_vertices);
+    std::iota(sources.begin(), sources.end(), 0);
+
+    if (num_src > 0 && num_src != num_vertices) {
+      std::mt19937 gen(rseed);
+      std::shuffle(sources.begin(), sources.end(), gen);
+      sources.resize(num_src);
     }
+  }
 
+  static galois::LargeArray<double>
+  compute(scea::graph::MutableGraph& g, std::vector<uint64_t> const& sources) {
```

Comment:
An unnecessary suggestion: default to all-source when an empty vector is passed as sources (so you don't have to make changes to the tests).

Reply:
Makes sense, will make this change.

Reply:
Unfortunately, since `sources` is a reference, it can't have a default argument. So either we force the vector to be copied/moved or we have to explicitly pass it in.

Reply:
I think the second is probably better.